core/src/main/java/com/hcl/ecomm/core/servlets/GetWishListItemsServlet.java

```diff
@@ -37,25 +40,28 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
 
          try {
             String customerToken = request.getHeader("CustomerToken");
-            JsonObject responseStream = wishListService.getWishListItems(customerToken);
+            JSONObject responseStream = getWishListItems(customerToken);
              LOG.debug("responseStream is {}", responseStream.toString());
-        
-            JsonArray wishlistitems= responseStream.get("items").getAsJsonArray();
+             JSONArray wishlistitems= responseStream.getJSONArray("items");
```

Comment:
Please apply null check here as well. @uditnaagar777

Reply:
Thanks Shubham, Done.

Reply:
if the response doesn't have "items", this will still result in NullPointerException.

Reply:
Thanks Shubham, I have updated this. pl check.